src/Fixer/Whitespace/StatementIndentationFixer.php

```diff
@@ -0,0 +1,604 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Analyzer\AlternativeSyntaxAnalyzer;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class StatementIndentationFixer extends AbstractFixer implements WhitespacesAwareFixerInterface
+{
+    private $bracesFixerCompatibility;
+
+    public function __construct(bool $bracesFixerCompatibility = false)
```

Comment:
It's a first place where we "configure" fixer via constructor parameter. I don't think it's good.
Also, the name doesn't say what behaviour it actually control - can you describe (in code, as variable name?) what is the behaviour is?

Also, why not pass it as fixer configuration?

Reply:
> I don't think it's good.

Why?

> Also, the name doesn't say what behaviour it actually control - can you describe (in code, as variable name?) what is the behaviour is?

`statement_indentation` has 3 specific cases where the behavior changes depending on this parameter. Would you like to have e.g. 1 parameter for each?

> Also, why not pass it as fixer configuration?

Because I don't like the `braces` specific behavior and I don't want it to remain once `braces` is removed. This means that the `$bracesFixerCompatibility` will be dropped along with `BracesFixer`. We would not be able to do that with a configuration exposed to end users.

---
Anyway, let's see the outcome of https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5960#discussion_r702463734 before acting here :)